src/condor_startd.V6/ResMgr.h

```diff
@@ -166,6 +166,11 @@ class BrokenItem {
 	std::string  b_tag;     // item tag, same as r_slot_id for broken slots
 	std::string  b_reason;  // reason string
 
+	// the broken id, converted to a sub_id suitable for use in an NFROwner struct
+	unsigned short int sub_id() const { return (unsigned short int)(0xFFFF - b_id); }
+	// convert back from a NFROwner dyn_id to a b_id
+	static unsigned int id_from_sub_id(unsigned int subid) { return subid - 0xFFFF; }
+
```

Comment:
Is a b_id value supposed to round-trip through these two functions? If so, then you need to change the expression in id_from_sub_id() to `0xFFFF - subid`.

Reply:
second function unused. I'll get rid of it.